lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -1578,15 +1578,19 @@ void ByteCodeGenerator::EmitScopeObjectInit(FuncInfo *funcInfo)
                     Assert(sym->GetScopeSlot() != Js::Constants::NoProperty && sym->GetScopeSlot() > slot);
                     propIds->elements[slot] = Js::Constants::NoProperty;
                 }
-                slot++;
             }
+            else
```

Comment:
Is there a way to do this only in case of pattern and not affect the other cases?